volume-pulseaudio/volume-pulseaudio

```diff
@@ -117,6 +120,11 @@ function print_format {
     echo "$1" | envsubst '${SYMB}${VOL}${INDEX}${NAME}'
 }
 
+function print_json {
+    echo "'{\"full_text\": \"$(print_format "$1")\", \"short_text\": \"$(print_format "$2")\", \"color\": \"$3\"}'"
```

Comment:
Json has special characters that need to be escaped. For instance, if the user's format string has a `"` in it, then this will likely crash the block. The way to solve this is with by writing a `json_escape` function that takes text and escapes it so that special characters are escaped, e.g. all `"` are replaced with `\"` and so on.

Reply:
I understand that random strings should be escaped, but having a proper parser in shell seems a bit too involved to me, especially since I don't believe the likelihood of having quotes coming from amixer is that high. Granted, users can configure the layout as they like, but again, getting quotes seems unlikely.

Anyway, I've removed all json formatting and I'll look into a different way of handling this, especially since in subscribe mode the blocklet doesn't get the events from the bar.

Reply:
Actually I think this was recently added that persistent blocks now do receive click events. This PR no longer does what the title claims, so I'm closing it. Feel free to make a new PR with your new solution.